src/env.rs

```diff
@@ -123,6 +123,22 @@ impl Rkv {
         })?;
         Ok(Store::new(db))
     }
+
+    /// Open an existing database, unlike other `open_or_create_*` functions, it
+    /// opens the given database by using a read transaction, which means other
+    /// in-flight write transaction will not block this call. This is preferred
+    /// to be used in the read_only scenarios.
+    pub fn open<'s, T, K>(&self, name: T) -> Result<Store<K>, StoreError>
+    where
+        T: Into<Option<&'s str>>,
+        K: AsRef<[u8]>,
+    {
+        let db = self.env.open_db(name.into()).map_err(|e| match e {
+            lmdb::Error::BadRslot => StoreError::open_during_transaction(),
```

Comment:
This isn't specific to your change, as it applies to the _Rkv::existing open_or_create*()_ methods, but I'm puzzled by the special handling of this error.

The lmdb-rs crate says of _Environment::open_db()_:

```
    /// This function will fail with `Error::BadRslot` if called by a thread which has an ongoing
    /// transaction.
```

And of _Environment::create_db()_:

```
    /// This function will fail with `Error::BadRslot` if called by a thread with an open
    /// transaction.
```

Which rkv translates into _StoreError::OpenAttemptedDuringTransaction_ ("attempted to open DB during transaction in thread").

But that seems to be what your test does, and it succeeds. Nor does _Rkv::existing open_or_create*()_ fail in that case; instead, it blocks. And we have no tests that trigger this error. So I wonder when we'll actually see it.

Reply:
This special error handling also puzzled me a lot. I think it has to do with [this](https://github.com/mozilla/rkv/blob/master/src/error.rs#L81). By default, `rkv::error` interprets it as `ReadTransactionAlreadyExists`, in this case, it wants to overwrite that with another interpretation.

I believe the `Error::BadRslot` is actually triggered by the transaction that being used to open the store. I managed to produce this error in this test:

```rs
 let _sk: Store<&str> = k.open_or_create("sk").expect("opened");
 let sk_readonly: Store<&str> = k.open("sk").expect("opened");
 let reader = sk_readonly.read(&k).expect("reader"); // this read transaction makes the following open panic with BadRslot
 let _sk_readonly_1: Store<&str> = k.open("sk").expect("opened");
```

Will add a test to cover this in a followup batch.